### PR TITLE
Auto approve dependabot PRs

### LINKED ISSUE #453: Dependabot automerge system
## 📰 Custom Issue

We are losing time on the Dependabot merge. These are PRs that are fine to be automated.

### PR DESCRIPTION
**Pull request by bjlittle**
### Description

This pull-request automates the maintenance task of approving and merging dependabot generated pull-requests.

Closes #453 

---

### CODE REVIEW

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Can we also implement auto MERGE?

**Comment by bjlittle** [comment]
@tkoyama010 There GHAs like https://github.com/pascalgn/automerge-action, which will auto-merge when a label is added. So, yes, I guess all possible.

Baby steps though ... let's try the auto-approve first and see how that goes. What do you think?

File: .github/workflows/ci-approve.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      pull-requests: write
+
+    if: github.actor == 'dependabot[bot]'
+    steps:
+      - name: "generate token"
+        uses: tibdex/github-app-token@v2
+        id: generate-token
+        with:
+          app_id: ${{ secrets.AUTH_APP_ID }}
+          private_key: ${{ secrets.AUTH_APP_PRIVATE_KEY }}
```
Comment (bjlittle) [inline .github/workflows/ci-approve.yml:25]: @tkoyama010 I have a GitHub App that generates tokens on request, so I'm curious to see whether chaining these two actions together works.

This pattern is also adopted in [ci-locks](https://github.com/bjlittle/geovista/blob/main/.github/workflows/ci-locks.yml#L109)

**Review by tkoyama010 — APPROVED** [review]
Great! Thanks.

**Pull request merged**